flowmachine/flowmachine/features/subscriber/active_subscribers.py

```diff
@@ -0,0 +1,181 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import date, datetime
+from typing import List, Union, Optional, Literal, NewType
+
+from flowmachine.core.mixins.exposed_datetime_mixin import ExposedDatetimeMixin
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.total_active_periods import (
+    TotalActivePeriodsSubscriber,
+)
+from flowmachine.features.utilities.events_tables_union import EventsTablesUnion
+import dateutil.rrule as rr
+from dateutil.relativedelta import relativedelta
+
+SubscriberSubsetType = NewType("SubscriberSubsetType", Union[str, list, Query, "Table"])
+
+
+class ActiveSubscribers(ExposedDatetimeMixin, Query):
+    """
+    Class that represents subscribers seen to be active.
+
+    To determine this, we regard the timeframe between `start_date` and `end_date` as the major period.
+    We then break the major period into `minor_period_count` minor periods.
```

Comment:
I'm getting a bit confused by all the periods here, but I don't think this docstring is quite describing the correct correspondence between the parameters and the query. We've got three different period lengths here:
- The innermost period ( = `minor_period_length` * `period_unit`),
- The middle-length period, which I think you're calling "minor period" here ( = `minor_period_count` * "innermost period"),
- The total period from `start_date` to `end_date`, which I think you're calling "major period".

On that basis, `minor_period_count` is the number of "innermost periods" per "minor period", and _not_ the number of "minor periods" per "major period". On the other hand, we could equally well call the innermost period "minor period" and the middle-length period "major period", in which case the description of `minor_period_count` here is correct but "we regard the timeframe between `start_date` and `end_date` as the major period" is incorrect.